### PR TITLE
8344060: Remove doPrivileged calls from shared implementation code in the java.desktop module : part 1

### PR DESCRIPTION
**Pull request by prsadhuk**
Since JEP 486 : Permanently Disable the Security Manager
[https://bugs.openjdk.org/browse/JDK-8338625] is now integrated, calls to java.security.AccessController.doPrivileged are obsolete and can be removed.

This PR takes care of some of the shared-platform files in the java.desktop module to have them removed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344060](https://bugs.openjdk.org/browse/JDK-8344060): Remove doPrivileged calls from shared implementation code in the java.desktop module : part 1 (**Bug** - P4)

### Reviewers
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) Review applies to [f27365fe](https://git.openjdk.org/jdk/pull/22133/files/f27365fea7ccae76f1f4a89a08e0f2b6b788fcdd)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22133/head:pull/22133` \
`$ git checkout pull/22133`

Update a local copy of the PR: \
`$ git checkout pull/22133` \
`$ git pull https://git.openjdk.org/jdk.git pull/22133/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22133`

View PR using the GUI difftool: \
`$ git pr show -t 22133`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22133.diff">https://git.openjdk.org/jdk/pull/22133.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22133#issuecomment-2478049364)
</details>

### CODE REVIEW

File: src/java.desktop/share/classes/sun/swing/SwingUtilities2.java [hunk 1]
```diff
@@ -1706,10 +1704,8 @@ private static Object makeIcon(final Class<?> baseClass,
                                   final String imageFile,
                                   final boolean enablePrivileges) {
         return (UIDefaults.LazyValue) (table) -> {
-            @SuppressWarnings("removal")
-            byte[] buffer = enablePrivileges ? AccessController.doPrivileged(
-                    (PrivilegedAction<byte[]>) ()
-                    -> getIconBytes(baseClass, rootClass, imageFile))
+            byte[] buffer = enablePrivileges ?
+                    getIconBytes(baseClass, rootClass, imageFile)
```
Comment (Glavo) [inline src/java.desktop/share/classes/sun/swing/SwingUtilities2.java:1708]:
```suggestion
            byte[] buffer = getIconBytes(baseClass, rootClass, imageFile);
```

Maybe you could move the method body into `makeIcon(Class<?>, Class<?>, String)`

File: src/java.desktop/share/classes/sun/java2d/opengl/OGLRenderQueue.java [hunk 2]
```diff
@@ -50,7 +48,7 @@ private OGLRenderQueue() {
          * The thread must be a member of a thread group
          * which will not get GCed before VM exit.
          */
-        flusher = AccessController.doPrivileged((PrivilegedAction<QueueFlusher>) QueueFlusher::new);
+        flusher = new QueueFlusher();
```
Comment (prrace) [inline src/java.desktop/share/classes/sun/java2d/opengl/OGLRenderQueue.java:51]: looks like you  forgot to remove SuppressWarnings

File: src/java.desktop/share/classes/sun/print/PSPrinterJob.java [hunk 3]
```diff
@@ -338,18 +338,11 @@ public class PSPrinterJob extends RasterPrinterJob {
         initStatic();
     }
 
-    @SuppressWarnings("removal")
     private static void initStatic() {
-       //enable privileges so initProps can access system properties,
+        //enable privileges so initProps can access system properties,
```
Comment (prrace) [inline src/java.desktop/share/classes/sun/print/PSPrinterJob.java:342]: This whole comment is obsolete.

File: src/java.desktop/share/classes/sun/print/RasterPrinterJob.java [hunk 4]
```diff
@@ -1008,19 +982,9 @@ public boolean printDialog(final PrintRequestAttributeSet attributes)
...
-                    PrintService service = getPrintService();
-                    if (service == null) {
-                        ServiceDialog.showNoPrintService(gc);
-                        return null;
-                    }
-                    return service;
-                }
-            });
-
+        PrintService service = getPrintService();
```
Comment (prrace) [inline src/java.desktop/share/classes/sun/print/RasterPrinterJob.java:985]: I'm fairly sure you missed removing a SuppressWarnings for this method

File: src/java.desktop/share/classes/sun/swing/JLightweightFrame.java [hunk 5]
```diff
@@ -105,24 +104,21 @@ public void updateCursor(JLightweightFrame frame) {
      * by the lock (managed with the {@link LightweightContent#paintLock()},
      * {@link LightweightContent#paintUnlock()} methods).
      */
-    @SuppressWarnings("removal")
-    private static boolean copyBufferEnabled = "true".equals(AccessController.
-            doPrivileged(new GetPropertyAction("swing.jlf.copyBufferEnabled", "true")));
+    private static boolean copyBufferEnabled = "true".equals(
+            System.getProperty("swing.jlf.copyBufferEnabled", "true"));
```
Comment (prrace) [inline src/java.desktop/share/classes/sun/swing/JLightweightFrame.java:108]: looks like you forgot to remove
import sun.security.action.GetPropertyAction;

File: src/java.desktop/share/classes/sun/swing/SwingUtilities2.java (hunk 1 above)
Comment (prrace) [reply src/java.desktop/share/classes/sun/swing/SwingUtilities2.java:1708]: This enablePrivileges param is interesting and obsolete.
Either here or in an immediate follow-on we should get rid of it.
I'm inclined to do it in a small follow-on, not here.
javax.swing.LookAndFeel.makeIcon(.) is the only place that calls
 SwingUtilities2.makeIcon_Unprivileged so it won't be hard.

File: src/java.desktop/share/classes/sun/awt/image/ImageWatched.java [hunk 6]
```diff
@@ -159,7 +135,7 @@ public boolean newInfo(Image img, int info,
             if (myiw == null) {
                 // My referent is null so we must prune in a second pass.
                 ret = true;
-            } else if (update(myiw, myref.acc, img, info, x, y, w, h) == false) {
+            } else if (update(myiw, img, info, x, y, w, h) == false) {
```
[the remark this answers is not in the document]
Comment (honkar-jdk) [reply src/java.desktop/share/classes/sun/awt/image/ImageWatched.java:138]: I agree, in-lining it as myiw.imageUpdate(img, info, x, y, w, h); is simpler.

**Review by honkar-jdk — COMMENTED** [review]
Went through the changes. Looks like few SM imports and SuppressWarnings removal were missed which @prrace has already mentioned.

File: src/java.desktop/share/classes/sun/awt/image/ImagingLib.java [hunk 7]
```diff
@@ -90,20 +88,10 @@ public static native int lookupByteRaster(Raster src, Raster dst,
...
-                    try {
-                        System.loadLibrary("mlib_image");
-                    } catch (UnsatisfiedLinkError e) {
-                        return Boolean.FALSE;
-                    }
-                    boolean success = init();
-                    return Boolean.valueOf(success);
-                }
-            };
+        System.loadLibrary("mlib_image");
```
[the remark this answers is not in the document]
Comment (prsadhuk) [reply src/java.desktop/share/classes/sun/awt/image/ImagingLib.java:91]: Other loadLibrary calls in other files in this PR were without try-catch so I removed it as I thought it redundant..
ANyway, I will revert it back..

File: src/java.desktop/share/classes/sun/awt/image/ImagingLib.java [hunk 8]
```diff
@@ -90,20 +88,14 @@ public static native int lookupByteRaster(Raster src, Raster dst,
...
-            };
+        boolean success = false;
+        try {
+            System.loadLibrary("mlib_image");
+            success = init();
+        } catch (UnsatisfiedLinkError e) {
+        }
 
-        useLib = AccessController.doPrivileged(doMlibInitialization);
+        useLib = Boolean.valueOf(success);
```
Comment (aivanov-jdk) [inline src/java.desktop/share/classes/sun/awt/image/ImagingLib.java:98]:
```suggestion
        useLib = success;
```
Unnecessary wrapping into `Boolean`.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

**Review by aivanov-jdk — APPROVED** [review]

File: src/java.desktop/share/classes/sun/swing/SwingUtilities2.java (hunk 1 above)
Comment (prrace) [reply src/java.desktop/share/classes/sun/swing/SwingUtilities2.java:1708]: I submitted https://bugs.openjdk.org/browse/JDK-8344569

**Review by prrace — APPROVED** [review]

File: src/java.desktop/share/classes/sun/swing/SwingUtilities2.java (hunk 1 above)
Comment (prsadhuk) [reply src/java.desktop/share/classes/sun/swing/SwingUtilities2.java:1708]: ok..you can assign to me if you want..

**Comment by prsadhuk** [comment]
Seems like there was a conflict for JLightweightFrame#initInterior with "8343476: Remove unnecessary @SuppressWarnings annotations (client)" commit..
Fixed but required a re-review

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**